simple/lint.go

```diff
@@ -1707,3 +1708,55 @@ func (c *Checker) LintRedundantSprintf(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintStringCopy(j *lint.Job) {
+	emptyStringLit := func(e ast.Expr) bool {
+		bl, ok := e.(*ast.BasicLit)
+		return ok && bl.Value == `""`
+	}
+	fn := func(node ast.Node) bool {
+		switch x := node.(type) {
+		case *ast.BinaryExpr: // "" + s, s + ""
+			if x.Op != token.ADD {
+				break
+			}
+			var want ast.Expr
+			switch {
+			case emptyStringLit(x.X):
+				want = x.Y
+			case emptyStringLit(x.Y):
+				want = x.X
+			default:
+				return true
+			}
+			j.Errorf(x, "should use %s instead of %s",
+				j.Render(want), j.Render(x))
+		case *ast.CallExpr: // string([]byte(s))
+			if len(x.Args) != 1 {
+				break
+			}
+			bt, ok := j.Program.Info.TypeOf(x.Fun).(*types.Basic)
+			if !ok || bt.Kind() != types.String {
+				break
+			}
+			nested, ok := x.Args[0].(*ast.CallExpr)
+			if !ok {
+				break
+			}
+			st, ok := j.Program.Info.TypeOf(nested.Fun).(*types.Slice)
+			if !ok {
+				break
+			}
+			et, ok := st.Elem().(*types.Basic)
+			if !ok || et.Kind() != types.Byte {
+				break
+			}
+			j.Errorf(x, "should use %s instead of %s",
```

Comment:
If the rendered form of `x` contains newlines, we should not emit the error. In the standard library, I found the following pattern:

```
r := reader("" + // for code formatting purpose.
	"a:\n" +
	" 0 \r\n" +
	"b:1 \t\r\n" +
	"c: 2\r\n" +
	" 3\t\n" +
	"  \t 4  \r\n\n")
```

where the `""` is only used to make the multi-line string easier to read. A image/jpeg test takes it to the extreme with the following:

```
const input = "" +
	"\xff\xd8\xff\xe0\x00\x10\x4a\x46\x49\x46\x00\x01\x01\x00\x00\x01" +
	"\x00\x01\x00\x00\xff\xdb\x00\x43\x00\x10\x0b\x0c\x0e\x0c\x0a\x10" +
	"\x0e\x89\x0e\x12\x11\x10\x13\x18\xff\xd8\xff\xe0\x00\x10\x4a\x46" +
	"\x49\x46\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00\xff\xdb\x00\x43" +
[...]
	"\xf9\x2e\xe0\x0a\x62\x7f\xdf\xd9"
```

It seems likely that the majority of multi-line strings with `""` concatenation in them are for formatting purposes.